### PR TITLE
Add a CSV preflight tutorial with local MCP integration tests

### PR DESCRIPTION
**Pull request by Congre123**
Add a runnable CSV preflight tutorial showing how to test a deterministic MCP tool through `Agent.call_tool` before attaching an LLM. It catches duplicate invoice IDs, non-finite amounts and invalid calendar dates, then returns a bounded report without invoice values. It uses synthetic fixtures, local stdio, and no provider API calls.

The example includes the validator, FastMCP server, programmatic application configuration, CLI, tutorial, and a gallery entry. It distinguishes failed tool execution from a successful validation that finds invalid data. It does not import data or change the SDK's runtime behavior.

Validation on Python 3.12.4/macOS, MCP SDK 1.30.0:

- All 9 tests pass against source `f62d849` and in a separate environment installed from the example requirements with published mcp-agent 0.2.6. Two tests use real local stdio MCP servers for discovery, invalid arguments, and valid/invalid fixture round trips. A CLI regression checks that an oversized file is rejected without reading its invalid UTF-8 tail.
- Ruff lint and format checks pass for the example. Scoped mypy passes with `--follow-imports=silent --ignore-missing-imports` for its three implementation files.
- CLI returns 0 for the valid fixture, 1 for the invalid fixture, and 2 for a missing file, including when launched outside the example directory.
- `git diff --check` passes. The full SDK suite, other Python versions/platforms, and model-provider calls were not run; this PR adds only an example and documentation.

AI disclosure: OpenAI Codex prepared the code and writing under @Congre123's authorization and executed the listed local checks. No human review or prior client work is claimed.

## Summary by CodeRabbit

- **New Features**
  - Added a CSV preflight example that validates invoice headers, unique IDs, decimal amounts, dates, formatting, and input-size limits.
  - Added command-line execution with JSON reports and distinct success, validation-failure, and execution-error statuses.
  - Added MCP integration for tool discovery and validation through an Agent without requiring an LLM.

- **Documentation**
  - Added setup, usage instructions, validation rules, expected outputs, and troubleshooting details.
  - Added the example to the integration gallery.

- **Tests**
  - Added coverage for valid files, edge cases, malformed input, error limits, size limits, CLI behavior, and MCP interactions.

### CODE REVIEW

File: examples/usecases/csv_preflight/main.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+
+def main() -> int:
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "csv_file", nargs="?", type=Path, default=EXAMPLE_DIR / "valid.csv"
+    )
+    args = parser.parse_args()
+    try:
+        report = asyncio.run(check_csv(args.csv_file.read_text(encoding="utf-8")))
```
[the remark this answers is not in the document]
Comment (Congre123) [reply examples/usecases/csv_preflight/main.py]: Addressed in 8e25e83. The CLI reads at most MAX_CHARS + 1 characters, preserving newlines, then lets the validator return the existing size-limit report. Added a real CLI regression with an oversized file and an invalid UTF-8 tail beyond the read limit. All nine tests pass against the source checkout and published mcp-agent 0.2.6; Ruff lint/format and scoped mypy pass.
